lib/hermetic/youtrack.ex

```diff
@@ -21,34 +21,92 @@ defmodule Hermetic.YouTrack do
   end
 
   @doc """
-    Return URL to YouTrack avatar for the given username.
+  Send HTTP PUT request to provided YouTrack endpoint.
+  """
+  def put!(endpoint) do
+    headers = [
+      {"authorization", "Bearer " <> token()},
+    ]
+
+    HTTPoison.put!(base_url() <> endpoint, "", headers)
+  end
+
+  @doc """
+  Send HTTP POST request to provided YouTrack endpoint.
+  """
+  def post!(endpoint) do
+    headers = [
+      {"authorization", "Bearer " <> token()},
+    ]
+
+    HTTPoison.post!(base_url() <> endpoint, "", headers)
+  end
+
+  @doc """
+  Create a new issue and return the issue id
+  """
+  @spec create_issue(String.t(), String.t(), String.t()) :: String.t()
+  def create_issue(project, summary, description) do
+    resp = put!("/rest/issue?" <> URI.encode_query([
+      project: String.upcase(project),
+      summary: summary,
+      description: description,
+    ]))
+    headers = Map.new(resp.headers)
+    headers["Location"] |> String.split("/") |> List.last
+  end
+
+  @doc """
+  Execute YouTrack command on an issue
+  """
+  @spec execute_command(String.t(), String.t()) :: HTTPoison.Response.t()
+  def execute_command(issue, command) do
+    post!("/rest/issue/#{issue}/execute?" <> URI.encode_query([
+      command: command,
+    ]))
+  end
+
+  @doc """
+  Return URL to YouTrack avatar for the given username.
   """
   def avatar_url(username) do
     base_url() <> request("/api/admin/users/#{username}?fields=avatarUrl")["avatarUrl"]
   end
 
   @doc """
-    Return URL to YouTrack logo.
+  Return URL to YouTrack logo.
   """
   def logo_url do
     base_url() <> "/static/apple-touch-icon-180x180.png"
   end
 
   @doc """
-    Fetch list of all available YouTrack project IDs.
+  Fetch list of all available YouTrack project IDs.
   """
   @spec project_ids :: list(String.t())
   def project_ids do
     for %{"shortName" => id} <- request("/rest/project/all"), do: id
   end
 
-  @spec project_ids :: list(String.t())
+  @spec cached_project_ids :: list(String.t())
   def cached_project_ids do
     Cache.get(YouTrack.ProjectIDs)
   end
 
+  @spec uncached_emails_to_logins :: %{String.t() => String.t()}
+  def uncached_emails_to_logins do
+    for %{"login" => login, "profile" => %{"email" => %{"email" => email}}} <-
+      request("/hub/rest/users?$top=999999999&fields=profile/email/email,login")["users"],
+      into: %{}, do: {email, login}
+  end
+
+  @spec emails_to_logins :: %{String.t() => String.t()}
+  def emails_to_logins do
```

Comment:
Be explicit about cached functions instead, those have weird properties like not returning the latest state that caller must know about.